src/cdk/drag-drop/drag-ref.ts

```diff
@@ -1238,13 +1238,22 @@ export class DragRef<T = any> {
       : point;
 
     if (this.lockAxis === 'x' || dropContainerLock === 'x') {
-      y = this._pickupPositionOnPage.y;
+      y =
+        this._pickupPositionOnPage.y -
+        (this.constrainPosition ? this._pickupPositionInElement.y : 0);
     } else if (this.lockAxis === 'y' || dropContainerLock === 'y') {
-      x = this._pickupPositionOnPage.x;
+      x =
+        this._pickupPositionOnPage.x -
+        (this.constrainPosition ? this._pickupPositionInElement.x : 0);
     }
 
     if (this._boundaryRect) {
-      const {x: pickupX, y: pickupY} = this._pickupPositionInElement;
+      // If not using a custom constrain we need to account for the pickup position in the element
```

Comment:
Can we capture this behavior in a unit test? That way we can ensure that it won't regress.

Reply:
I have now added a unit test showing that the boundary constraining to an element stays the same when using `constrainPosition` as it is without using it.